File: mobile-android-java-rent/app/src/main/java/ru/rbkdev/lease/account/login/CLoginActivity.java
package ru.rbkdev.lease.account.login;

import ru.rbkdev.lease.R;

import ru.rbkdev.lease.main.CID;
import ru.rbkdev.lease.main.CMainActivity;
import ru.rbkdev.lease.account.login.sign.CSignActivity;
import ru.rbkdev.lease.account.login.create.CCreateActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.io.File;

public class CLoginActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.account_login_activity);

        Button btnCreateNow = findViewById(R.id.btnCreateNow);
        Button btnSignAccount = findViewById(R.id.btnSignAccount);
        TextView txtMessage = findViewById(R.id.txtAccountLoginMessage);
        txtMessage.setText("");

        Intent intent = getIntent();
        if(intent != null) {

            int result = intent.getIntExtra(CID.INTENT_RESULT, 0);
            if(result != 0) {

                String message = intent.getStringExtra(CID.INTENT_MESSAGE);
                if(message != null)
                    txtMessage.setText(message);

                if(result == -1 || result == 2) {
                    btnCreateNow.setVisibility(View.INVISIBLE);
                    btnSignAccount.setVisibility(View.INVISIBLE);
                }
            }
        }

        btnSignAccount.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(CLoginActivity.this, CSignActivity.class);
                startActivityForResult(intent, CID.RESULT_RETURN_SIGN);
            }
        });

        btnCreateNow.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(CLoginActivity.this, CCreateActivity.class);
                startActivityForResult(intent, CID.RESULT_RETURN_CREATE);
            }
        });
    }

    @Override
    public void onBackPressed() {
        setResult(RESULT_OK);
        finish();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
        super.onActivityResult(requestCode, resultCode, intent);

        String valueIntent;

        if(resultCode == RESULT_OK) {

            File file = new File(getExternalFilesDir(null), CID.FILE_NAME);
            if(file.delete() || !file.delete()) {
                if(requestCode == CID.RESULT_RETURN_CREATE) {
                    valueIntent = intent.getStringExtra(CID.INTENT_USER_MAIL);
                    CMainActivity.RefreshUserId(file, "AM" + valueIntent);
                    setResult(RESULT_OK);
                    finish();
                }

                if(requestCode == CID.RESULT_RETURN_SIGN) {
                    valueIntent = intent.getStringExtra(CID.INTENT_USER_ID);

                    CMainActivity.RefreshUserId(file, valueIntent);
                    finish();
                }
            }
        }
    }
}

File: mobile-android-kotlin-rent-clean/app/src/main/java/ru/rbkdev/rent/clean/ui/login/CLoginActivity.kt
package ru.rbkdev.rent.clean.ui.login

import ru.rbkdev.rent.clean.R

import ru.rbkdev.rent.clean.CClient
import ru.rbkdev.rent.clean.CDataExchange
import ru.rbkdev.rent.clean.ui.CMainActivity
import ru.rbkdev.rent.clean.CSettings

import android.os.Bundle
import android.widget.Toast
import android.widget.Button
import android.widget.TextView
import android.content.Intent

import androidx.appcompat.app.AppCompatActivity
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts

import com.yandex.mapkit.MapKitFactory

/***/
class CLoginActivity : AppCompatActivity() {

    private var mResultLauncher: ActivityResultLauncher<Intent>? = null

    /***/
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.login_activity)

        val txtLoginLogin: TextView = findViewById(R.id.txtLoginLogin)
        val txtLoginPassword: TextView = findViewById(R.id.txtLoginPassword)
        val btnLoginEntry: Button = findViewById(R.id.btnLoginEntry)

        var login = CSettings.getInstance().getLogin(baseContext)
        if (login != "null")
            txtLoginLogin.text = login

        var password = CSettings.getInstance().getPassword(baseContext)
        if (password != "null")
            txtLoginPassword.text = password

        btnLoginEntry.setOnClickListener {

            login = txtLoginLogin.text.toString()
            password = txtLoginPassword.text.toString()

            if (password == "ip") {
                val array = login.split(":")
                if (array.size == 2) {
                    CSettings.getInstance().setLinkPreference(baseContext, array[0], array[1])
                    Toast.makeText(baseContext, "success", Toast.LENGTH_SHORT).show()
                    return@setOnClickListener
                }
            }

            val dataExchange = CDataExchange()
            dataExchange.setCommand("login_clean")
            dataExchange.setIdUser("0")
            dataExchange.setIdHouse("0")
            dataExchange.setMessage("${txtLoginLogin.text}:${txtLoginPassword.text}")

            CClient.startExchange(baseContext, dataExchange)
            if (dataExchange.getCode() == R.string.code_success) {

                CSettings.getInstance().setUserId(dataExchange.getIdUser())
                CSettings.getInstance().setLoginPreference(baseContext, login, password)

                mResultLauncher?.launch(Intent(baseContext, CMainActivity::class.java))
            } else
                Toast.makeText(baseContext, "Неправильный ввод", Toast.LENGTH_SHORT).show()
        }

        mResultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {

            finish()

            MapKitFactory.setApiKey("MapKey")
        }
    }
}